src/main/java/io/aiven/kafka/connect/gcs/TopicPartitionRecordGrouper.java

```diff
@@ -83,24 +84,27 @@ public void put(final SinkRecord record) {
 
         final TopicPartition tp = new TopicPartition(record.topic(), record.kafkaPartition());
         final SinkRecord currentHeadRecord = currentHeadRecords.computeIfAbsent(tp, ignored -> record);
-        final String filename = renderFilename(tp, currentHeadRecord);
+        final String filename = renderFilename(tp, currentHeadRecord).render();
 
         if (shouldCreateNewFile(filename)) {
             // Create new file using this record as the head record.
             currentHeadRecords.put(tp, record);
-            final String newFilename = renderFilename(tp, record);
+            final String newFilename = renderFilename(tp, record).render();
             fileBuffers.computeIfAbsent(newFilename, ignored -> new ArrayList<>()).add(record);
         } else {
             fileBuffers.computeIfAbsent(filename, ignored -> new ArrayList<>()).add(record);
         }
     }
 
-    private String renderFilename(final TopicPartition tp, final SinkRecord headRecord) {
+    protected boolean acceptsTemplateInstance(final Template filenameTemplate) {
+        return TopicPartitionRecordGrouper.acceptsTemplate(filenameTemplate);
+    }
+
+    protected Instance renderFilename(final TopicPartition tp, final SinkRecord headRecord) {
```

Comment:
Why did you move the render out of this function? Every caller stills calls render on it anyway.

If you don't want to perform the render inside this method, the name does not make sense anymore. Maybe sth. like `getBoundFilenameTemplate` (java names rock 😄 )

Reply:
I need to move render out because this method is overriden in the HourlyTopicPartitionRecordGrouper, where super is used to call render on this class, and then this render is extended.

Reply:
He did it for the extension in the class `HourlyTopicPartitionRecordGrouper`: there there is `return super.renderFilename(tp, headRecord)` call, result of which is later extended by hourly params.

However, this is a pretty poor way of extension, that harms readability of the code. 

Instead of removing the `.render()` call from the `renderFilename` method - which makes its name wrong - please, take the pre-`.render()` lines and extract them into a different method, like `bindFilenameTemplate` or something, close to what @cedrics wrote and then keep `.render` in the `renderFilename` method, instead of having it in every single caller.

Reply:
ok maybe it is better in the terms of naming, I renamed it